pkg/local_object_storage/engine/inhume.go

```diff
@@ -227,6 +234,23 @@ func (e *StorageEngine) inhumeAddr(addr oid.Address, prm shard.InhumePrm) (bool,
 
 	prm.SetTargets(append(children, addr)...)
 
+	if shardWithObject != "" {
+		sh := e.getShard(shardWithObject)
+
+		_, err := sh.Inhume(prm)
+		if err != nil {
+			if !errors.Is(err, logicerr.Error) {
+				e.reportShardError(hashedShard(sh), "could not inhume object in shard", err)
+			}
+
+			return false, err
+		}
+
+		return true, nil
```

Comment:
How about root/children?

Reply:
root and children cases do not fill shard ID cause they do not have a shard, every shard can have info about them, this is handled below